Mage/src/main/java/mage/abilities/common/BoastAbility.java

```diff
@@ -0,0 +1,82 @@
+package mage.abilities.common;
+
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.ActivatedAbilityImpl;
+import mage.abilities.condition.Condition;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.Costs;
+import mage.abilities.costs.mana.ManaCosts;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.Effects;
+import mage.constants.EffectType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.watchers.common.AttackedThisTurnWatcher;
+
+/**
+ *
+ * @author weirddan455
+ */
+public class BoastAbility extends ActivatedAbilityImpl {
+
+    private static final Effects emptyEffects = new Effects();
+
+    public BoastAbility(Effect effect, ManaCosts cost) {
+        super(Zone.BATTLEFIELD, effect, cost);
+        this.maxActivationsPerTurn = 1;
+        this.addWatcher(new AttackedThisTurnWatcher());
+        this.condition = BoastAbilityCondition.instance;
+    }
+
+    public BoastAbility(Effect effect, Cost cost) {
+        super(Zone.BATTLEFIELD, effect, cost);
+        this.maxActivationsPerTurn = 1;
+        this.addWatcher(new AttackedThisTurnWatcher());
+        this.condition = BoastAbilityCondition.instance;
+    }
+
+    public BoastAbility(Effect effect, Costs<Cost> costs) {
+        super(Zone.BATTLEFIELD, effect, costs);
+        this.maxActivationsPerTurn = 1;
+        this.addWatcher(new AttackedThisTurnWatcher());
+        this.condition = BoastAbilityCondition.instance;
+    }
+
+    public BoastAbility(BoastAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public Effects getEffects(Game game, EffectType effectType) {
```

Comment:
I know this is in the `ConditionalActivatedAbility` code but I'm honestly not sure what it's supposed to do there and it can probably be removed.